src/cloudant/views.py

```diff
@@ -325,3 +334,94 @@ def custom_result(self, **options):
         rslt = self.make_result(**options)
         yield rslt
         del rslt
+
+class QueryIndexView(View):
+    """
+    A view that defines a JSON index in a design document.  
+
+    If you wish to manage a view that represents a query index it is strongly
+    recommended that :func:`~cloudant.database.CloudantDatabase.create_index`
+    and :func:`~cloudant.database.CloudantDatabase.delete_index` are used.
+    """
+    def __init__(self, ddoc, view_name, map_fields, reduce_func, **kwargs):
+        if not isinstance(map_fields, dict):
+            raise CloudantArgumentError('The map property must be a dictionary')
+        if not isinstance(reduce_func, basestring):
+            raise CloudantArgumentError('The reduce property must be a string.')
+        super(QueryIndexView, self).__init__(
+            ddoc,
+            view_name,
+            map_fields,
+            reduce_func,
+            **kwargs
+        )
+        self['map'] = map_fields
+        self['reduce'] = reduce_func
+        self.result = None
+
+    @property
+    def map(self):
+        """
+        Provides a map property accessor and setter.
+
+        :param dict map_func: A dictionary of fields defining the index.
+
+        :returns: Fields defining the index
+        """
+        return self.get('map')
+
+    @map.setter
+    def map(self, map_func):
+        """
+        Provides a map property setter.
+        """
+        if isinstance(map_func, dict):
+            self['map'] = map_func
+        else:
+            raise CloudantArgumentError('The map property must be a dictionary')
+
+    @property
+    def reduce(self):
+        """
+        Provides a reduce property accessor and setter.
+
+        :param str reduce_func: A string representation of the reduce function
+            used in part to define the index.
+
+        :returns: Reduce function as a string
+        """
+        return self.get('reduce')
+
+    @reduce.setter
+    def reduce(self, reduce_func):
+        """
+        Provides a reduce property setter.
+        """
+        if isinstance(reduce_func, basestring):
+            self['reduce'] = reduce_func
+        else:
+            raise CloudantArgumentError('The reduce property must be a string')
+
+    def __call__(self, **kwargs):
+        """
+        QueryIndexView objects are not callable.  If you wish to execute a query
+        using a query index, use
+        :func:`~cloudant.database.CloudantDatabase.get_query_result` instead.
+        """
+        raise CloudantException(
+            'A QueryIndexView is not callable.  If you wish to execute a query '
+            'use the database \'get_query_result\' convenience method.'
+        )
+
+    def make_result(self, **options):
```

Comment:
Guess this is overriding a base method that doesnt make sense for this object, would maybe spell that out in the docstring.